### ISSUE TITLE
Null image ID on import for scaleway_instance_server after migration from C2M to GP1-XS

### ISSUE THREAD

**Issue by sefam-mchatelain** [scaleway/terraform-provider-scaleway#705]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.29 + provider.scaleway v1.16.0

### Affected Resource(s)

Instance ID = 910733b9-e363-4ade-92a4-8a4a8996ee0b.

The instance has been migrated from `C2M` to `GP1-XS` following [this procedure][proc] (with snapshot).
After the migration, when checking the instance information, both field "From image" and "Image ID" are missing (they are empty). Other fields are correct.

  [proc]: https://www.scaleway.com/en/docs/migrate-c2-arm64-to-virtual-instance-using-rsync/

### Terraform Configuration Files

I import the new resource into the existing Terraform configuration with

```console
$ terraform import scaleway_instance_server.fr_sefam_lab fr-par-1/910733b9-e363-4ade-92a4-8a4a8996ee0b
```
and adding the resource in `scaleway_resources.tf` :

```hcl
# scaleway_resources.tf
resource "scaleway_instance_server" "fr_sefam_lab" { }
```

### Debug Output
`terraform.tfstate` reports a null image ID that was imported for the resource.

```hcl
# terraform.tfstate
...
    "id": "fr-par-1/910733b9-e363-4ade-92a4-8a4a8996ee0b",
    "image": null,
...
```

### Expected Behavior

Image ID should be imported, since it is a mandatory parameter to declare the resource in Terraform afterwards.

```hcl
# scaleway_resources.tf
resource "scaleway_instance_server" "fr_sefam_lab" { 
  type    = "GP1-XS"
  image   = <insert-image-id-here>
  state   = "started"
}
```

### Actual Behavior

Using `null` returns an error. Using other image ID like:

+ the one from the snapshot (`c1593629-3eff-4883-a3aa-8362f58555f1`
+ the one for debian-stretch (`6c261f2a-71e1-4280-8c75-223962ecf6dc`) 
+ the one from other similar instances (`5f7ab0cd-d10e-49a0-91c8-315ed8a6a1a4`)
leads to replacement, which is obviously not what I want.

```console
  # scaleway_instance_server.fr_sefam_lab must be replaced
-/+ resource "scaleway_instance_server" "fr_sefam_lab" {
...
      ~ id                               = "fr-par-1/910733b9-e363-4ade-92a4-8a4a8996ee0b" -> (known after apply)
      + image                            = "5f7ab0cd-d10e-49a0-91c8-315ed8a6a1a4" # forces replacement
```

### Steps to Reproduce

1. import the resource
2. `terraform plan` with different IDs for the field `image`

### Important Factoids

+ importing other resources (volumes, ips) using a similar procedure works as intended.
+ I was send here by the Scaleway support, although I do not think it is an issue with the plugin itself, but rather with the attribution of an image ID during the migration process. The plugin correctly reports a `null` ID as there seems to be no image ID attached to my instance.

### References

* #0000

**Comment by remyleone** [comment]
Hello @sefam-mchatelain, have you tried making an image out of the snapshot? You can click on "Image from snapshot" in the scaleway console (https://console.scaleway.com/instance/snapshots). Once it is done, you should have an image id. 
As for the import of a server from a snapshot, we are going to investigate that with the provider.

**Comment by sefam-mchatelain** [comment]
Hi @remyleone thanks for the quick reply, I did not try that yet. I'll give it a shot later today and report back asap.

**Comment by sefam-mchatelain** [comment]
Reporting back: creating an image from the snapshot indeed provides an image ID. But planning still indicates that the resource "must be replaced". 
```console
  # scaleway_instance_server.fr_sefam_lab must be replaced
-/+ resource "scaleway_instance_server" "fr_sefam_lab" {
...
      ~ id                               = "fr-par-1/910733b9-e363-4ade-92a4-8a4a8996ee0b" -> (known after apply)
      + image                            = "8084fbc2-1ee0-423a-8540-1c01ad07f568" # forces replacement
        ip_id                            = "fr-par-1/14658a6b-f76a-4d96-b616-55f6793d2ea6"
...
```
I assume that means I would loose all changes that occurred since the snapshot was created (about 10 days ago), right?

**Comment by remyleone** [comment]
yes, it would probably mean that the server would be replaced. One option could be to create a new snapshot with the recent changes you did, then creating an image from this snapshot to start a valid server with a valid image you want. Is this solution acceptable for you?

**Comment by sefam-mchatelain** [comment]
Indeed, I reached the same conclusion. I'll plan that, and report back next week. Better having 2h downtime than loosing 10 days of work ;) Thanks for the help!

**Comment by sefam-mchatelain** [comment]
*Reporting back:* converting the snapshot into an image fixed the importing issue, instance now has the image-ID properly defined. And `terraform plan` only updates the instance, instead of replacing it. Great.

I'm left with one question then: now that image and instance are "linked", I guess I must maintain this image forever? Because I end up with both an image and a volume that are stored (and billed) on my account (and that partially contain the same information).

Further investigation suggest that I should rather convert the snapshot into a volume, create an "empty" instance based on an image-ID provided by [Scaleway end-point][sep], detach the default volume, and attach "my" volume to it instead. Then I would only have an instance, a volume, and no image to maintain. Correct?

And any news on your side regarding the image-ID not properly set-up when using a snapshot?
Cheers.

  [sep]: https://api-marketplace.scaleway.com/images?page=1&per_page=100

**Comment by remyleone** [comment]
I'm not sure about your particular setup but I was able to import an image using the following code:

```hcl
terraform {
  required_providers {
    scaleway = {
      source  = "scaleway/scaleway"
      version = "2.2.1-rc.3"
    }
  }
}

provider "scaleway" {
}

data "scaleway_instance_image" "name" {
  name = "Ubuntu 20.04 Focal Fossa"
  zone = "fr-par-1"
}

data "scaleway_instance_ip" "name" {
  address = "51.15.248.118"
}

resource "scaleway_instance_server" "main" {
  type = "GP1-S"
  image = data.scaleway_instance_image.name.image_id
  enable_ipv6 = true
  ip_id = data.scaleway_instance_ip.name.id
}
```

Beware as images are zoned. So for terraform to find the right image you would need to setup the correct zone and image name. In case you have a new issue do not hesitate to open a new issue with a minimal snippet of code that reproduces your problem.
